bin/bootstrap_practice_exercise.exs

```diff
@@ -0,0 +1,258 @@
+    lower_camel_case
+    |> String.split(~r/(?=[A-Z\d])/)
+    |> Enum.map_join("_", &String.downcase/1)
+  end
+
+  def get_properties(%{"cases" => cases}) when is_list(cases) do
+    Enum.map(cases, &Generate.get_properties/1)
+    |> Enum.concat()
+    |> Enum.uniq()
+  end
+
+  def get_properties(%{"property" => property, "input" => %{} = input}),
+    do: [
+      {property,
+       Enum.map(input, fn
+         {var, %{} = val} -> {var, Enum.map(val, fn {v, _} -> v end)}
+         {var, _} -> {var, nil}
+       end)}
+    ]
+
+  def get_properties(%{"property" => property}), do: [{property, "input"}]
+  def get_properties(_data), do: []
+
+  def print_property({property, variables}) do
+    """
+     @doc \"\"\"
+     insert function description here
+     \"\"\"
+     @spec #{property}(#{Enum.map_join(variables, ", ", fn
+      {var, nil} -> "#{var} :: String.t()"
+      {var, sub_vars} -> "#{var} :: %{#{Enum.map_join(sub_vars, " : String.t(), ", fn v -> v end)} : String.t()}"
+    end)}) :: {:ok, String.t()} | {:error, String.t()}
+     def #{property}(#{Enum.map_join(variables, ", ", fn {var, _} -> var end)}) do
+     end
+    """
+  end
+
+  def print_comments(comments, do_not_print) do
+    comments
+    |> Enum.reject(fn {field, _value} -> field in do_not_print end)
+    |> Enum.map_join("\n", fn
+      {"comments", values} when is_list(values) ->
+        "# #{Enum.map_join(values, "\n# ", &String.trim/1)}"
+
+      {field, values} when is_list(values) ->
+        "#\n# --#{field} --\n# #{Enum.map_join(values, "\n# ", &inspect/1)}"
+
+      {field, value} ->
+        "#\n# -- #{field} --\n# #{inspect(value)}"
+    end)
+  end
+
+  def print_input(%{} = input) do
+    Enum.map_join(input, "\n", fn {variable, value} -> "#{variable} = #{inspect(value)}" end)
+  end
+
+  def print_input(input), do: "input = #{inspect(input)}"
+
+  def print_variables(%{} = input),
+    do: Enum.map_join(input, ", ", fn {variable, _value} -> variable end)
+
+  def print_variables(input), do: input
+
+  def print_expected(%{"error" => err}), do: "{:error, #{inspect(err)}}"
+  def print_expected(expected), do: "{:ok, #{inspect(expected)}}"
+
+  def print_test_case(%{"description" => description, "cases" => sub_cases} = c, module) do
+    """
+    describe \"#{description}\" do
+    #{Generate.print_comments(c, ["description", "cases"])}
+    #{Enum.map_join(sub_cases, "\n\n", &Generate.print_test_case(&1, module))}
+    end
+    """
+  end
+
+  def print_test_case(
+        %{
+          "description" => description,
+          "property" => property,
+          "input" => input,
+          "expected" => expected
+        } = c,
+        module
+      ) do
+    """
+    @tag :pending
+    test \"#{description}\" do
+      #{Generate.print_comments(c, ["description", "property", "input", "expected", "uuid"])}
+      #{print_input(input)}
+      output = #{module}.#{Generate.to_snake_case(property)}(#{print_variables(input)})
```

Comment:
It would be good if changing to snake case happened as soon as possible, right after fetching and decoding the JSON, and not when printing the test case. For example, you forgot to use it when printing the stub 😉 so for an exercise like `allergies` (https://github.com/exercism/problem-specifications/blob/e2779fc84ff07cb5bd8ff89f38e78af6d17b5056/exercises/allergies/canonical-data.json#L14), you get a stub like this:
```elixir
def allergicTo(item, score) do
end
```

but a function call in tests like this:
```elixir
output = Allergies.allergic_to(item, score)
```

It's also necessary to change properties to snake case, for example: https://github.com/exercism/problem-specifications/blob/daf620d47ed905409564dec5fa9610664e294bde/exercises/all-your-base/canonical-data.json#L20-L22